source/MaterialXRuntime/Private/PvtNodeGraph.cpp

```diff
@@ -72,8 +72,9 @@ void PvtNodeGraph::addPort(PvtDataHandle portdef)
     const PvtPortDef* pd = portdef->asA<PvtPortDef>();
     const uint32_t flags = pd->getFlags();
 
-    PvtNode::Port p;
-    p.value = pd->getValue();
+    PvtNode::PortData p;
+    p.value = RtValue::createNew(pd->getType(), this->getObject());
+    RtValue::copy(pd->getType(), pd->getValue(), p.value);
```

Comment:
Fix proper allocation and copy of value. Needed if the value holds "large" data that needs to be allocated and stored as pointer (e.g. string, filename, Matrix44 etc.)